### PR TITLE
#165383302 Implement delete comments functionality

### PR DESCRIPTION
**Pull request by TheDrizzyWay**
#### Description
This PR adds a functionality which enables users delete their comments

#### Type of change
- New Feature

#### How Has This Been Tested?
- Unit tests were written and run
- The feature was tested on postman

#### Checklist:
- [x] Write unit tests for this feature
- [x] Create a controller for deleting comments
- [x] Update swagger document to include this feature
#### Story (PT)
[#165383302](https://www.pivotaltracker.com/n/projects/2320805/stories/165383302)

### CODE REVIEW

File: server/routes/profile.routes.js [hunk 1]
```diff
@@ -3,19 +3,20 @@ import { ProfileController } from '../controllers';
 import { signInAuth } from '../utils/users/permissions.util';
 import {
   tryCatch,
-  idValidator,
+  UuidValidator,
```
Comment (Chiazokam) [inline server/routes/profile.routes.js]: Hi @TheDrizzyWay , isn't there a way to validate the uuid with validator.js?

File: server/controllers/comment.controller.js [hunk 2]
```diff
@@ -126,6 +126,41 @@ class CommentController {
...
+    const commentExists = await table.findOne({
+      where: {
+        id: commentId,
+        userId: id
+      }
+    });
+
+    if (!commentExists) return Response(res, 404, 'Comment not found.');
+
+    await table.destroy({
```
Comment (geekdanitek) [inline server/controllers/comment.controller.js:181]: I would like clarity for code written on line `155` and `158`

Comment (TheDrizzyWay) [reply server/controllers/comment.controller.js:181]: We are deleting both the comment and all its associated thread comments. Hence the need for two calls.

Comment (Chiazokam) [reply server/controllers/comment.controller.js:181]: Will DELETE ON CASCADE not work here?

Comment (chukwudidu) [reply server/controllers/comment.controller.js:181]: @TheDrizzyWay making an extra call is quite expensive. Referencing the column on the migrations file as @Chiazokam suggested should fix this.

File: server/controllers/comment.controller.js [hunk 3]
```diff
@@ -126,6 +126,41 @@ class CommentController {
...
+      where: { id: commentId }
+    });
+    await table.destroy({
+      where: { parentId: commentId }
+    });
```
Comment (samorahno) [inline server/controllers/comment.controller.js:186]: @TheDrizzyWay good work, but is there any way you can make the table destroy call once?

Comment (TheDrizzyWay) [reply server/controllers/comment.controller.js:186]: The comment and thread comments are not linked via foreign key, so two queries are needed.

Comment (chukwudidu) [reply server/controllers/comment.controller.js:186]: @TheDrizzyWay Oh, I see that now. 👍

**Review by chukwudidu — APPROVED** [review]

**Review by davidokonji — APPROVED** [review]

File: server/controllers/comment.controller.js (hunk 2 above)
Comment (TheDrizzyWay) [reply server/controllers/comment.controller.js:181]: There's no foreign key linking the two parties involved.

**Review by BukkyOmo — APPROVED** [review]

**Review by davidokonji — APPROVED** [review]

**Review by geekdanitek — APPROVED** [review]

**Review by paradisekelechi — APPROVED** [review]

**Review by samorahno — APPROVED** [review]

**Review by Chiazokam — APPROVED** [review]

**Pull request merged**